src/daemon/client.c

```diff
@@ -273,6 +277,82 @@ client_handle_get_interfaces(struct lldpd *cfg, enum hmsg_type *type,
 	return output_len;
 }
 
+#ifdef ENABLE_LLDPMED
+/**
+ * Set local chassis inventory info
+ * Input: chassis object
+ * Output: updated chassis object
+ */
+static ssize_t
+client_handle_set_local_chassis(struct lldpd *cfg, enum hmsg_type *type,
+    void *input, int input_len, void **output, int *subscribed)
+{
+	struct lldpd_chassis *chassis = NULL;
+	struct lldpd_chassis *local_chassis = NULL;
+	struct utsname un;
+
+	log_debug("rpc", "client request a change in chassis configuration");
+	if (lldpd_chassis_unserialize(input, input_len, &chassis) <= 0) {
+		*type = NONE;
+		return 0;
+	}
+
+	local_chassis = LOCAL_CHASSIS(cfg);
+
+	free(local_chassis->c_med_hw);
+	local_chassis->c_med_hw = (!chassis->c_med_hw) ? dmi_hw() : strdup(chassis->c_med_hw);
+
+	// Follows lldpd.c - only set sw if advertising is enabled
+	if (cfg->g_config.c_advertise_version) {
+		free(local_chassis->c_med_sw);
+
+		if (!chassis->c_med_sw) {
+			if (uname(&un) < 0) {
+				log_warn("rpc", "Could not get default uname. Will continue anyway.");
+				local_chassis->c_med_sw = NULL;
+			}
+			else
+				local_chassis->c_med_sw = strdup(un.release);
+		}
+		else
+			local_chassis->c_med_sw = strdup(chassis->c_med_sw);
```

Comment:
I think this change will get erased at regular interval if you dojn't prevent that in `lldpd.c`.

Reply:
How do you prefer to achieve this?

I can think of a flag to check if it was manually set, if so, don't update

Why is software revision updated periodically in the first place? Not like sw will change without rebooting whatever machine it is running on....
- could kernel live-patching change software revision without reboot?

Reply:
Not updating software revision periodically would be the simpler workaround I think.

Reply:
How about moving this code structure to, https://github.com/lldpd/lldpd/blob/45f3c77332be56ce2df3bcf8c402bd1f14d29f39/src/daemon/lldpd.c#L1159
so it is called only once?

Reply:
Yes, this seems a good idea.